subsys/bluetooth/common/Kconfig

```
@@ -244,6 +244,10 @@ config BT_RPA
 	bool
 	depends on BT_HOST_CRYPTO || BT_CTLR_CRYPTO
 
+config BT_PRIVATE_SHELL
+	bool
+	default n
```

May I suggest an alternative here?

```suggestion
config BT_PRIVATE_SHELL
	def_bool BT_SHELL || BT_MESH_SHELL 
```

`select` should be used sparingly: https://docs.zephyrproject.org/latest/build/kconfig/tips.html#select-pitfalls, and while this specific use is one of the "safe" uses of `select`, defaulting to the values that would otherwise select it is more flexible (works with ints too), and avoids any future issues with `select`, while achieving the same. 

That being said, it could also be argued that `BT_PRIVATE_SHELL` is a bit overkill, and that 

```
zephyr_library_sources_ifdef(CONFIG_BT_PRIVATE_SHELL bt_shell_private.c)
```

could just be modified to 
```
if (CONFIG_BT_SHELL OR CONFIG_BT_MESH_SHELL)
	zephyr_library_sources(bt_shell_private.c)
endif()
```

If you want to keep it as is, then I my only suggestion would be to
```suggestion
config BT_PRIVATE_SHELL
	bool
```

as `bool` Kconfigs always default to false.

---

Thanks for your comment. May I also get a suggestion from @PavelVPV?
Additionally, I'm waiting for the CI to turn green first.

---

If you want to keep Kconfig option, then I think it is exactly the case for `select`. It won't cause issues described in the doc as these functions don't bring new dependencies.

It can be a bit more clear to have a Kconfig option rather than including `bt_shell_private.c` by explicitly checking bt shell o rmesh shell Kconfig options state as it won't require CMakeLists.txt change if these functions need to be enabled in some other cases. But I leave it to you here.